examples/nucleo_f767zi/ethernet/main.cpp

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (c) 2020, Mike Wolfram
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <modm/board.hpp>
+#include <modm/driver/ethernet/lan8720a.hpp>
+#include <modm/processing/rtos.hpp>
+#include <FreeRTOS_IP.h>
+#include <FreeRTOS_Sockets.h>
+
+using namespace Board;
+
+namespace Ethernet
+{
+	using RMII_Ref_Clk = GpioInputA1;
+	using RMII_Mdio = GpioA2;
+	using RMII_Crs_Dv = GpioInputA7;
+	using RMII_Tx_En = GpioOutputG11;
+	using RMII_Tx_D0 = GpioOutputG13;
+	using RMII_Tx_D1 = GpioOutputB13;
+	using RMII_Mdc = GpioOutputC1;
+	using RMII_Rx_D0 = GpioInputC4;
+	using RMII_Rx_D1 = GpioInputC5;
+	using Port = Eth<modm::Lan8720a>;
+}
+
+UBaseType_t ulNextRand;
+
+void vApplicationIPNetworkEventHook(eIPCallbackEvent_t eNetworkEvent);
+
+class NetworkInitTask : modm::rtos::Thread
+{
+public:
+	NetworkInitTask()
+	: Thread(5, 2048, "network_init")
+	{}
+
+	void
+	run()
+	{
+		uint8_t ipAddress[4] { 192, 168, 1, 1 };
+		uint8_t netmask[4] { 255, 255, 255, 0 };
+		uint8_t gatewayAddress[4] { 0, 0, 0, 0 };
+		uint8_t dnsAddress[4] { 0, 0, 0, 0 };
+
+		// local MAC address
+		uint8_t macAddress[] { 0x02, 0x00, 0x00, 0x00, 0x00, 0x00 };
+
+		// A real MAC address can be retrieved from the Microchip 24AA02E48
+		// I2C EEPROM, which is locaed at address 0xFA.
+
+		// initialize random numbers
+	    time_t now;
+	    time(&now);
+	    ulNextRand = uint32_t(now);
```

Comment:
We could have used real random numbers here...

```suggestion
		// initialize random numbers
	    modm::platform::RandomNumberGenerator::enable();
	    while(!modm::platform::RandomNumberGenerator::isReady()) {}
	    ulNextRand = modm::platform::RandomNumberGenerator::getValue();
```

Reply:
Yes, and as a fallback you can seed srand with the device's UID.

Reply:
Also `time()` is not implemented. We could perhaps wire it to `modm::Clock`, but it's not done yet.

Reply:
> Yes, and as a fallback you can seed srand with the device's UID.

I am not sure if this is a good idea or a serious security flaw...

Reply:
Well, you need to collect entropy somehow. There are a bunch more hardware calibration registers that provide entropy within some bounds. It's a somewhat good idea since the values will be different for every device (assuming a good hash function), however, only if the attacker cannot read them out remotely. However, security is definitely beyond the scope of modm and too much work to implement yourself. We'd have to integrate some embedded TLS library for encryption.